xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/test/java/org/xwiki/rendering/internal/resolver/RelativeResourceReferenceEntityReferenceResolverTest.java

```diff
@@ -0,0 +1,408 @@
+            (Answer<Boolean>) invocation -> existingDocuments.contains(invocation.getArguments()[0]));
+
+        when(this.defaultEntityProvider.getDefaultReference(EntityType.DOCUMENT))
+            .thenReturn(new EntityReference(DEFAULT_PAGE, EntityType.DOCUMENT));
+    }
+
+    private DocumentResourceReference documentResource(String referenceString, boolean typed)
+    {
+        DocumentResourceReference reference = new DocumentResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+
+    private SpaceResourceReference spaceResource(String referenceString, boolean typed)
+    {
+        SpaceResourceReference reference = new SpaceResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+
+    private AttachmentResourceReference attachmentResource(String referenceString, boolean typed)
+    {
+        AttachmentResourceReference reference = new AttachmentResourceReference(referenceString);
+
+        reference.setTyped(typed);
+
+        return reference;
+    }
+    // Tests
+
+    @Test
+    void resolve()
+    {
+        assertNull(this.resolver.resolve(null, null));
+        assertNull(this.resolver.resolve(new ResourceReference("path", ResourceType.PATH), null));
+    }
+
+    @Test
+    void resolveTypeDocument()
+    {
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE),
+            this.resolver.resolve(documentResource(WIKI + ':' + SPACE + '.' + PAGE, true), null));
+        assertEquals(PAGE_ENTITY_REFERENCE, this.resolver.resolve(documentResource(SPACE + '.' + PAGE, true), null));
+
+        assertEquals(new EntityReference(PAGE, EntityType.DOCUMENT),
+            this.resolver.resolve(documentResource(PAGE, true), null));
+
+        assertNull(this.resolver.resolve(documentResource("", true), null));
+
+        ResourceReference withBaseReference = documentResource("", true);
+        withBaseReference.addBaseReference(WIKI + ':' + SPACE + '.' + PAGE);
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE),
+            this.resolver.resolve(withBaseReference, null));
+
+        assertEquals(new DocumentReference(WIKI, SPACE, PAGE), this.resolver
+            .resolve(documentResource("", true), null, new DocumentReference(WIKI, SPACE, PAGE)));
+    }
+
+    @Test
+    void resolveUntypeDocument()
+    {
+        // When the page does not exist
+
+        assertEquals(new DocumentReference(WIKI, List.of(SPACE, PAGE), DEFAULT_PAGE),
+            this.resolver.resolve(documentResource(WIKI + ':' + SPACE + '.' + PAGE, false), null));
+
+        // TODO: check that this assert is ok, we don't get the default page since we don't have any info to properly
+        // resolve the absolute ref (no wiki, no base reference)
+        assertEquals(PAGE_ENTITY_REFERENCE,
+            this.resolver.resolve(documentResource(SPACE + '.' + PAGE, false), null));
+        assertEquals(PAGE_ALONE_ENTITY_REFERENCE,
```

Comment:
Now, one problem is that adding a `WebHome` in the result of that example where you have no space is that it completely change the behavior of the relative -> absolute resolution (as soon as a reference contains a space, it's considered "absolute" in a given wiki, which is not the intent here).

Reply:
> it completely change the behavior of the relative -> absolute resolution

I'm trying to understand what you mean exactly here. My understanding is that you mean there's an inconsistency between:
  1. Resolving a ResourceReference `Space.WebHome` to an absolute EntityReference which would result in `currentwiki.CurrentSpace.Space.WebHome`
  2. Resolving a ResourceReference `Space.WebHome` to a relative EntityReference which would result in `Space.WebHome` and then resolving that one to an absolute EntityReference which would result in `defaultwiki:Space.WebHome`

Am I understanding it correctly?

Reply:
I think you understood, but your examples are not accurate since the `ResourceReference` is `Space`.

Reply:
There's actually a 3rd solution: we accept the inconsistency of getting different results, depending if you resolved directly the ResourceReference to absolute or if you resolved it first to relative and then the EntityReference to absolute